### ISSUE TITLE
Scientific tools packaged in a docker container create theirs files as root.

### ISSUE THREAD

**Issue by prioux** [aces/cbrain#631]
This is a limitation of docker, but for CBRAIN we need to handle this.

The .dockerjob.sh file (see also #630) should include a bit of code that runs something like this, after the scientific task is done:

```bash
retcode=$? # preserve previosu return code
find /PATH/TO/TASKWORKDIR -xdev -owner root -print0 | xargs -0 chown CBRAIN_UID:CBRAIN_GID
exit $retcode
```

where to stuff in uppercase is substituted by CBRAIN itself with proper values. Care must be taken such that the return code of the previous command is preserved.

This is not perfect, as the docker container might have been configured to run under any number of UIDs. The sysadmin must double check that before even deploying the tool.

**Comment by glatard** [comment]
The right way to do this I think is to start the container as user `cbrain`, using `-u $UID:$GID`. I think this is how it was initially implemented in CBRAIN. The Docker documentation for CBRAIN used to say that the container is supposed to run under any user.

**Comment by prioux** [comment]
Yes, excellent, that `-u` option is ideal. I hope it's supported by Guillimin's `udocker` though. Trying to get an interactive node now.

**Comment by prioux** [comment]
We're lucky that udocker supports `-u` (even if it ignores it completely).

I'm going to fix the code and also introduce some safety check within the .docker.job file to verify that we are running under the proper UID and GID even before launching the scientific task.

**Comment by prioux** [comment]
Fixed by #634
